.github/workflows/pytest-remote-data.yml

```diff
@@ -0,0 +1,115 @@
+#
+# We prevent this by only running this workflow when a maintainer has looked
+# over the PR's diff and verified that nothing malicious seems to be going on.
+# The maintainer then adds the "remote-data" label to the PR, which will then
+# trigger this workflow via the combination of the "on: ... types:"
+# and "if:" sections below.  The first restricts the workflow to only run when
+# a label is added to the PR and the second requires one of the PR's labels
+# is the "remote-data" label.  Technically this is slightly different from
+# triggering when the "remote-data" label is added, since it will also trigger
+# when "remote-data" is added, then later some other label is added.  Maybe
+# there's a better way to do this.
+#
+# But wait, you say!  Can't a malicious PR get around this by modifying
+# this workflow file and removing the label requirement?  I think the answer
+# is "no" as long as we trigger the workflow on "pull_request_target" instead
+# of the usual "pull_request".  The difference is what context the workflow
+# runs inside: "pull_request" runs in the context of the fork, where changes
+# to the workflow definition will take immediate effect, while "pull_request_target"
+# runs in the context of the main pvlib repository, where the original (non-fork)
+# workflow definition is used instead.  Of course by switching away from the fork's
+# context to keep our original workflow definitions, we're also keeping all the
+# original code, so the tests won't be run against the PR's new code.  To fix this
+# we explicitly check out the PR's code as the first step of the workflow.
+# This allows the job to run modified pvlib & pytest code, but only ever via
+# the original workflow file.
+# So long as a maintainer always verifies that the PR's code is not malicious prior to
+# adding the label and triggering this workflow, I think this should not present
+# a security risk.
+#
+# Note that this workflow can be triggered again by removing and re-adding the
+# "remote-data" label to the PR.
+#
+# Note also that "pull_request_target" is also the only way for the secrets
+# to be accessible in the first place.
+# 
+# Further reading:
+# - https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+# - https://github.community/t/can-workflow-changes-be-used-with-pull-request-target/178626/7
+
+name: pytest-remote-data
+
+on:
+  pull_request_target:
+    types: [labeled]
+  push:
+    branches:
+    - master
+
+jobs:
+  test:
+    env:
+      # we need to turn "3.6" into 36 to build the conda requirement filename,
+      # but there's no good way to do that via string manipulation.  So set up
+      # a map that will be parsed as JSON later:
+      VERSIONMAP: '{"3.6": "36", "3.7": "37", "3.8": "38", "3.9": "39"}'
+
+    strategy:
+      fail-fast: false  # don't cancel other matrix jobs when one fails
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
```

Comment:
Do we need to run remote data on all versions? I think min and max are likely fine but also fine to let the computers chug.